src/main/java/calculator/StringParser.java

```diff
@@ -0,0 +1,53 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringParser {
+    private static final String DEFAULT_DELIMITER = ",:";
+    private static final Pattern PATTERN = Pattern.compile("(//(.+)\\n)?(.*)");
+
+    private StringParser() {
+        throw new IllegalStateException("utility 입니다.");
+    }
+
+    public static List<Integer> parse(String numberSequence) {
+        Matcher matcher = PATTERN.matcher(numberSequence);
+        String delimiters = generateDelimiter(matcher);
+        String numbers = matcher.group(3);
+
+        return Arrays.stream(numbers.split(delimiters))
+                .map(StringParser::parseElement)
+                .collect(Collectors.toList());
+    }
+
+    private static String generateDelimiter(Matcher matcher) {
+        StringBuilder sb = new StringBuilder(DEFAULT_DELIMITER);
+
+        if (matcher.find() && Objects.nonNull(matcher.group(2))) {
+            sb.append(matcher.group(2));
+        }
+
+        return new StringBuilder("[]")
+                .insert(1, sb)
+                .toString();
```

Comment:
문자열 연산이 많은 경우라면 StringBuilder 객체를 생성해서 문자열 연산을 수행할 수도 있겠지만, 
지금의 경우에는 불필요하게 StringBuilder 객체를 생성하지 않고 아래와 같이 바로 문자열 연산을 수행하는 것이 더 효율적인 코드가 될 것 같아요. 🤔

```suggestion
        return "[" + sb + "]"
```